src/toolbar_editor.cpp

```diff
@@ -248,13 +252,18 @@ void Toolbar_Editor::insert_action(QAction *new_action)
 
 void Toolbar_Editor::on_button_remove_toolbar_clicked()
 {
-    toolbar_items.remove(combo_toolbar->currentText());
+    QString name = combo_toolbar->currentText();
+
+    if (!allowToolbarRemoval(name))
+        return;
+
+    toolbar_items.remove(name);
     combo_toolbar->removeItem(combo_toolbar->currentIndex());
 }
 
 void Toolbar_Editor::on_button_add_toolbar_clicked()
 {
-    QString name = QString("custom_toolbar_%1").arg(toolbar_items.size());
+    QString name = customToolbarNamePrefix + QString::number(toolbar_items.size());
```

Comment:
The diff is just that I use the `customToolbarNamePrefix` variable instead of the text `custom_toolbar_` directly. But you are right, your current implementation breaks custom toolbar names if someone removes a toolbar. I have no easy fix for this without storing the maximum number to the settings or parse all custom toolbars (which would be doable).